internal/healthcheck/healthcheck.go

```diff
@@ -233,3 +255,34 @@ func getNodeStatusCondition(conditions []corev1.NodeCondition, conditionType cor
 
 	return false
 }
+
+func IsSriovNetworkNodeReady(ctx context.Context, c client.Reader, l logr.Logger) error {
+	// Check the CRD first. If it doesn't exist, pass the health check
+	crd := &apiextensionsv1.CustomResourceDefinition{}
+	if err := c.Get(ctx, types.NamespacedName{Name: "sriovnetworknodestates.sriovnetwork.openshift.io"}, crd); err != nil {
+		if errors.IsNotFound(err) {
+			return nil
+		}
+
+		return fmt.Errorf("failed to verify sriov CRD exists: %w", err)
+	}
+
+	nodeStates := &sriovv1.SriovNetworkNodeStateList{}
+
+	if err := c.List(ctx, nodeStates); err != nil {
+		return fmt.Errorf("failed to get SriovNetworkNodeStates: %w", err)
+	}
+
+	if len(nodeStates.Items) == 0 {
+		return fmt.Errorf(SriovNetworkNodeStateNotPresentMsg)
```

Comment:
Worth logging that we are doing the special workaround in case devs (I) fall for this (i had crd but no resources!)